src/ring_theory/localization/away.lean

```diff
@@ -194,3 +196,200 @@ lemma is_localization.away.finite_presentation (r : R) {S} [comm_ring S] [algebr
+      map_one],
+    refine nonpos_of_neg_nonneg (le_of_lt _),
+    rwa [neg_neg, ← not_le] },
+end
+
+lemma self_zpow_neg_mul (d : ℤ) : self_zpow x B (-d) * self_zpow x B d = 1 :=
+by rw [mul_comm, self_zpow_mul_neg x B d]
+
+-- noncomputable!
+-- lemma self_zpow_invertible (d : ℤ) : invertible (self_zpow x B d ) :=
+-- ⟨self_zpow x B (-d), self_zpow_neg_mul x B d, self_zpow_mul_neg x B d⟩
+
+lemma self_zpow_pow_sub (a : R) (b : B) (m d : ℤ) :
+  (self_zpow x B (m - d)) * mk' B a (1 : submonoid.powers x) = b ↔
+  (self_zpow x B m) * mk' B a (1 : submonoid.powers x) = (self_zpow x B d) * b :=
+begin
+  rw [sub_eq_add_neg, self_zpow_add, mul_assoc, mul_comm _ (mk' B a 1), ← mul_assoc],
+  split,
+  { intro h,
+    have := congr_arg (λ s : B, s * self_zpow x B d) h,
+    simp only at this,
+    rwa [mul_assoc, mul_assoc, self_zpow_neg_mul, mul_one, mul_comm b _] at this},
+  { intro h,
+    have := congr_arg (λ s : B, s * self_zpow x B (-d)) h,
+    simp only at this,
+    rwa [mul_comm _ b, mul_assoc b _ _, self_zpow_mul_neg, mul_one] at this}
+end
+
+-- lemma pow_eq_algebra_map (d : ℕ) :
+--   (((self_as_unit x B)^(d : ℤ) : Bˣ) : B) = (algebra_map R B x)^d :=
+-- by simp only [self_as_unit, zpow_coe_nat, units.coe_pow, units.coe_mk]
+
+variables [is_domain R] [normalization_monoid R] [unique_factorization_monoid R]
+-- include hx
+
+-- theorem exists_reduced_fraction {b : B} (hb : b ≠ 0) :
+--   ∃ (a : R) (n : ℤ), ¬ x ∣ a ∧
+--     (((self_as_unit x B)^n : Bˣ) : B) * mk' B a (1 : submonoid.powers x) = b :=
+-- begin
+--   classical,
+--   obtain ⟨⟨a₀, y⟩, H⟩ := surj (submonoid.powers x) b,
+--   obtain ⟨d, hy⟩ := (submonoid.mem_powers_iff y.1 x).mp y.2,
+--   have ha₀ : a₀ ≠ 0,
+--   { haveI := @is_domain_of_le_non_zero_divisors B _ R _ _ _ (submonoid.powers x) _
+--       (powers_le_non_zero_divisors_of_no_zero_divisors hx.ne_zero),
+--     simp only [map_zero, ← subtype.val_eq_coe, ← hy, map_pow] at H,
+--     apply ((injective_iff_map_eq_zero' (algebra_map R B)).mp _ a₀).mpr.mt,
+--     rw ← H,
+--     apply mul_ne_zero hb (pow_ne_zero _ _),
+--     exact is_localization.to_map_ne_zero_of_mem_non_zero_divisors B
+--       (powers_le_non_zero_divisors_of_no_zero_divisors (hx.ne_zero))
+--       (mem_non_zero_divisors_iff_ne_zero.mpr hx.ne_zero),
+--     exact is_localization.injective B (powers_le_non_zero_divisors_of_no_zero_divisors
+--       (hx.ne_zero)) },
+--   simp only [← subtype.val_eq_coe, ← hy] at H,
+--   obtain ⟨m, a, hyp1, hyp2⟩ := max_power_factor ha₀ hx,
+--   refine ⟨a, m-d, _⟩,
+--   rw [self_as_unit_pow_sub x B a b m d, pow_eq_algebra_map x B d, mul_comm _ b, ← map_pow, H, hyp2,
+--     pow_eq_algebra_map x B m, map_mul, map_pow],
+--   exact ⟨hyp1, (congr_arg _ (is_localization.mk'_one _ _))⟩,
+-- end
+
+
+theorem exists_reduced_fraction' {b : B} (hb : b ≠ 0) (hx : irreducible x) :
+  ∃ (a : R) (n : ℤ), ¬ x ∣ a ∧
+    (self_zpow x B n) * mk' B a (1 : submonoid.powers x) = b :=
```

Comment:
The existence of [is_localization.mk'_one](https://leanprover-community.github.io/mathlib_docs/ring_theory/localization/basic.html#is_localization.mk'_one) suggests to rewrite to:
```suggestion
    self_zpow x B n * algebra_map R B a = b :=
```
On the other hand, it is not a `@[simp]` lemma so I'm happy to keep this as-is.